components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -0,0 +1,326 @@
+#include "catch2/catch.hpp"
+
+#include "physics/share/physics_constants.hpp"
+
+#include "share/util/scream_common_physics_functions.hpp"
+
+#include "ekat/ekat_pack.hpp"
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+#include "ekat/util/ekat_test_utils.hpp"
+
+namespace{
+
+template<typename ScalarT, int NumLevels>
+struct ChecksHelpers {
+
+  static bool is_non_negative (const ScalarT& s, const int k) {
+    return not ( k<NumLevels && (s<0 || isnan(s)) );
+  }
+  static bool equal (const ScalarT& lhs, const ScalarT& rhs) {
+    return lhs==rhs;
+  }
+  static bool approx_equal (const ScalarT lhs, const ScalarT rhs, 
```

Comment:
Neither here nor there, but I've used `nearly_equal` for this type of function. Same number of letters without abbreviation.